src/esm.ts

```diff
@@ -65,11 +66,23 @@ export function registerAndCreateEsmHooks(opts?: RegisterOptions) {
 
     // pathname is the path to be resolved
 
-    return nodeResolveImplementation.defaultResolve(
+    console.log('got here');
+
+    const x = nodeResolveImplementation.defaultResolve(
       specifier,
       context,
       defaultResolve
     );
+    console.log('x', x);
+    return x;
+  }
+
+  async function load(
+    url: string,
+    context: {},
+    defaultLoad: typeof load
+  ): Promise<{ format: Format }> {
+    throw new Error('load');
```

Comment:
With the new loader hook consolidation, either the `resolve` or the `load` hook can fill the role of `getFormat`.  I haven't had time to dig in; do you know if it's better for `resolve` or `load` to perform `getFormat` duties?  I assume it should be combined with `resolve`, since there is nothing in `load` that affects the format.

Reply:
I'm not sure yet but I'll look deeper into how to merge the hooks. Just need to get load to be called at all first (node's current nightly is not calling into it for some reason).

Reply:
Ah, forgot the re-export at the root of the repo (esm.mjs). Now load is getting called :-).

Reply:
According to docs load() should return an object with format and source. So I think `resolve` can be left untouched and `load` should just call into the existing `getFormat` and `transformSource`. The main problem is the default that is being to deferred to is different as it is for `load`. So I need to find some way to provide a default for the existing getFormat` and `transformSource`.

Reply:
Another thing to get right:

What happens when an ESM module tries to import a CommonJS module?  And what if that CommonJS module is `.ts` and needs to be transpiled?  I assume, although I'm not sure, that `defaultLoad` will call out to our `require.extensions` hook if you pass it a URL pointing to a TS file and `format: 'commonjs'`

Reply:
I did some experimenting.  Here is what I found:

If `defaultLoad` does not receive a `format`, it will internally call the old `defaultGetFormat`.  If the computed format is `commonjs`, it will *not* attempt to load anything.  Instead it will return `{format: 'commonjs', source: null}`.  This tells node to use the CommonJS codepath, which respects our `require.extensions` hooks.

The problem is: if the format is determined to be `'module'`, it will try to load it.  If the file doesn't exist it will error.  Today, we intentionally call `defaultGetFormat` with files that do not exist.  If we want to classify `foo.ts`, we call `defaultGetFormat` with `foo.js`.

The situation is no better if *both* `foo.js` and `foo.ts` exist.  Because then it's loading the wrong file.  And there are other formats: json, wasm, etc.

Reply:
Should we begin shipping our own copy of `defaultGetFormat`?

I believe that this is the correct file, at the correct version number matching what we already have:

https://github.com/nodejs/node/blob/v15.3.0/lib/internal/modules/esm/get_format.js

Matches the version from which we pulled https://github.com/TypeStrong/ts-node/blob/main/dist-raw/node-esm-resolve-implementation.js (See comments at top of file)

Reply:
I added the file you suggested and patched its imports and it seems to work.